pkg/controller/config/config_controller.go

```diff
@@ -197,13 +217,35 @@ func (r *ReconcileConfig) Reconcile(ctx context.Context, request reconcile.Reque
 
 	r.cacheManager.ExcludeProcesses(newExcluder)
 	configSourceKey := aggregator.Key{Source: "config", ID: request.NamespacedName.String()}
+	// if gvksToSync is empty, cachemanager will CancelData for stale expectations
 	if err := r.cacheManager.UpsertSource(ctx, configSourceKey, gvksToSync); err != nil {
+		// for readiness, try to cancel the expectations on error for the gvks
+		for _, gvk := range gvksToSync {
+			u := &unstructured.Unstructured{}
+			u.SetGroupVersionKind(gvk)
+
+			if cfgTracker.TryCancelExpect(u) {
```

Comment:
This doesn't make sense.  cfgTracker is the OT for config objects, which will never contain a sync source gvk like "pod"

Reply:
Chatted offline that config is a little different, even today:

https://github.com/open-policy-agent/gatekeeper/blob/366ba791d009c70f18f85fd94dbd5aed8426fc9d/pkg/readiness/ready_tracker.go#L720-L729

Subsequent work may move to correct this "hack" that can be done where an OT tracks more than 1 gvk